src/main/kotlin/com/sparetimedevs/suspendmongo/crud/InternalOperation.kt

```diff
@@ -18,20 +18,24 @@ package com.sparetimedevs.suspendmongo.crud
 
 import com.mongodb.reactivestreams.client.Success.SUCCESS
 import com.sparetimedevs.suspendmongo.Collection
+import com.sparetimedevs.suspendmongo.resilience.Resilience
 import com.sparetimedevs.suspendmongo.result.Error
 import com.sparetimedevs.suspendmongo.result.Result
 import kotlinx.coroutines.reactive.awaitFirstOrNull
 import kotlinx.coroutines.reactive.consumeEach
 import org.bson.conversions.Bson
+import java.util.function.Supplier
 
 //TODO create better error messages.
 
+private val resilience: Resilience by lazy { Resilience() }
```

Comment:
Don’t think this is a smart way to initialise a dependency. But maybe I lag knowledge regarding lazy.
In any case, I do not see how a user of the library could set the Resilience with different properties. Maybe this dependency could be part of the Database object. Which can be created by the user of the library. That way the user does have control over how resilience the database calls need to be.

Reply:
`class Database(connectionString: String, name: String, val resilience: Resilience = Resilience()) { ... }`

Reply:
Check out this for more info regarding delegated properties. [how-kotlins-delegated-properties-and-lazy-initialization-work](https://medium.com/til-kotlin/how-kotlins-delegated-properties-and-lazy-initialization-work-552cbad8be60)

I'll take a look at refactoring to allow override of the resilience properties as per your suggestion. I'll push another update later!

Reply:
Thanks for the article, good read. I guess it is a viable option if we want to initialise Resilience with default values only (for now).